azurerm_notification_hub/main.tf

```diff
@@ -0,0 +1,62 @@
+provider "azurerm" {
+  version = "~>1.43"
+}
+
+terraform {
+  backend "azurerm" {}
+}
+
+data "azurerm_key_vault" "key_vault" {
+  name                = var.key_vault_name
+  resource_group_name = var.resource_group_name
+}
+data "azurerm_key_vault_secret" "notification_hub_gcm_key" {
+  name         = "notification-hub-01-gc-m-key"
+  key_vault_id = data.azurerm_key_vault.key_vault.id
+}
+data "azurerm_key_vault_secret" "notification_hub_bundle_id" {
+  name         = "notification-hub-01-bundle-id"
+  key_vault_id = data.azurerm_key_vault.key_vault.id
+}
+data "azurerm_key_vault_secret" "notification_hub_key_id" {
+  name         = "notification-hub-01-key-id"
+  key_vault_id = data.azurerm_key_vault.key_vault.id
+}
+data "azurerm_key_vault_secret" "notification_hub_team_id" {
+  name         = "notification-hub-01-team-id"
+  key_vault_id = data.azurerm_key_vault.key_vault.id
+}
+data "azurerm_key_vault_secret" "notification_hub_token_id" {
+  name         = "notification-hub-01-token"
+  key_vault_id = data.azurerm_key_vault.key_vault.id
+}
+
+# New infrastructure
+
+resource "azurerm_notification_hub_namespace" "notification_hub_ns" {
+  name                = local.ntfns_resource_name
+  resource_group_name = var.resource_group_name
+  location            = var.region
+  namespace_type      = var.ntfns_namespace_type
+  sku_name            = var.ntfns_sku_name
+  enabled             = var.ntfns_enabled
+}
+
+resource "azurerm_notification_hub" "notification_hub" {
+  name                = local.resource_name
+  namespace_name      = azurerm_notification_hub_namespace.notification_hub_ns.name
+  resource_group_name = var.resource_group_name
+  location            = var.region
+
+  apns_credential {
```

Comment:
the apns_credential  is optional  according to the documentation: https://www.terraform.io/docs/providers/azurerm/r/notification_hub.html#apns_credential
it looks like you always use that block.
you might try to use an object like the one  @francescopersico already asked me to use:
eg:
```
variable backup_configuration {
  type = object({
    frequency = number
    max_snapshot_count = number
    storage_connection_string = string
  })
  default = null
}
```
main.tf
```
......
redis_configuration {
    enable_authentication         = var.enable_authentication
    rdb_backup_enabled            = var.backup_configuration != null
    rdb_backup_frequency          = var.backup_configuration != null ? var.backup_configuration.frequency : null
    rdb_backup_max_snapshot_count = var.backup_configuration != null ? var.backup_configuration.max_snapshot_count : null
    rdb_storage_connection_string = var.backup_configuration != null ? var.backup_configuration.storage_connection_string : null
  }
.....
```